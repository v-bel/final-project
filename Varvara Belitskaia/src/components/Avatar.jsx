/* eslint-disable import/prefer-default-export */
import React from 'react';

export const Avatar = ({ poke, isForInfoPage }) => {
  const src = `https://raw.githubusercontent.com/epam-js-dec-2018/final-project/master/pokemons/${
    poke.id
  }.png`;
  const altText = `${poke.name}`;
  const classNameValue = isForInfoPage
    ? 'pokemon-info-avatar card-img-top'
    : 'pokemon-avatar';
  return (
    <img
      src={src}
      alt={altText}
      className={classNameValue}
      onError={e => {
        e.target.onerror = null;
        e.target.src = 'src/assets/default.png';
      }}
    />
  );
};
