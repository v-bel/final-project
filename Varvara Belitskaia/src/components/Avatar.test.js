import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Avatar } from './Avatar';

const poke = {
  name: 'bulbasaur',
  id: 1
};

it(`should render the pokemon's avatar`, () => {
  const enzymeWrapper = shallow(<Avatar poke={poke} />);
  expect(toJson(enzymeWrapper)).toMatchSnapshot();
});
