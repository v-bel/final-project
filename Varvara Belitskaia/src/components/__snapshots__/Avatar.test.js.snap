// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the pokemon's avatar 1`] = `
<img
  alt="bulbasaur"
  className="pokemon-avatar"
  onError={[Function]}
  src="https://raw.githubusercontent.com/epam-js-dec-2018/final-project/master/pokemons/1.png"
/>
`;
